src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -833,6 +835,7 @@ private static void CleanupSendState(State state)
 
             if (!MsQuicStatusHelper.SuccessfulStatusCode(status))
             {
+                HandleWriteCompletedState();
```

Comment:
How would this help? The HandleWriteCompletedState does nothing if the state is `Pending`

Reply:
I don't know what I was thinking, should be fixed now. Thanks for catching it.